lib/streamlit/connections/snowpark_connection.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# NOTE: We won't always be able to import from snowflake.snowpark.session so need the
+# `type:ignore` comment below, but that comment will explode if `warn-unused-ignores` is
+# turned on when the package is available. Unfortunately, mypy doesn't provide a good
+# way to configure this at a per-line level :(
+# mypy: no-warn-unused-ignores
+
+import configparser
+import os
+import threading
+from contextlib import contextmanager
+from datetime import timedelta
+from typing import TYPE_CHECKING, Any, Dict, Iterator, Optional, Union, cast
+
+import pandas as pd
```

Comment:
nit: for optimizing performance ([context](https://github.com/streamlit/streamlit/issues/6066)), we might remove all top-level imports of pandas. So it could be better here to move this under the `if TYPE_CHECKING` section.

Reply:
done!

Reply:
Hm, actually I'll leave this as-is for now because pandas is used for reasons other than type-checking (we need the `read_sql` function).

We still may eventually want to only import pandas before it's used for speeding up start times, but I think it's less clear that we want to do this right now when there's non-typing module usage going on.